Riot/Modules/LocationSharing/LocationManager.swift

```diff
@@ -191,7 +194,14 @@ extension LocationManager: CLLocationManagerDelegate {
     func locationManagerDidChangeAuthorization(_ manager: CLLocationManager) {
         
         let status = self.locationManager.authorizationStatus
-        self.authorizationRequestDidComplete(with: status)
+        authorizationReturnedSinceRequestingAlways = true
+        if status == .authorizedAlways {
```

Comment:
What if the user do not authorize "always"?
My guess is that the completion wouldn't be called and the caller wouldn't show the alert to go to the settings.
wdyt?

Reply:
Yes with these changes the alert to go to the settings is only shown the second time the user selects Live Location Sharing when the time timeout method is used(as iOS does not let you request the permission a second time.)

It's going to be difficult to show the settings alert directly after  "Keep Only While Using" is selected (if we even want that?) as the comment [here](https://github.com/vector-im/element-ios/pull/7223/files#diff-442d76e1c62c8c08c1a92587405f9865a47411e0c16b1c7149d82f8b4c3b9aceR199) suggests `locationManagerDidChangeAuthorization` can get called multiple times. So there is not an easy way to know the user has actually selected "Keep Only While Using" vs `locationManagerDidChangeAuthorization` being called before the first user prompt is show.

I think these changes substantially improve the user experience either way, so I would be in favour of merging this and then if we decide it's necessary to show the settings alert directly after "Keep Only While Using" selection we can look for a separate solution for that.

Reply:
This behavior sometimes breaks the expectation of the caller having the completion always called.
But I understand your reasoning and I think this is a good compromise for now. 👍🏻